crates/graphql/src/schema.rs

```diff
@@ -919,12 +926,20 @@ impl QueryRoot {
             },
         );
 
-        let rows: Vec<models::Nft> = query
+        let mint_addresses_subquery = token_accounts::table
+            .select(token_accounts::mint_address)
             .filter(
-                metadatas::address.eq(any(metadata_creators::table
-                    .select(metadata_creators::metadata_address)
-                    .filter(metadata_creators::creator_address.eq(any(creators))))),
-            )
+                token_accounts::owner_address
+                    .eq(any(owners.unwrap_or_else(Vec::new)))
+                    .and(token_accounts::amount.eq(1)),
+            );
+        let metadata_addresses_subquery = metadata_creators::table
+            .select(metadata_creators::metadata_address)
+            .filter(metadata_creators::creator_address.eq(any(creators.unwrap_or_else(Vec::new))));
+
+        let rows: Vec<models::Nft> = query
+            .filter(metadatas::address.eq(any(metadata_addresses_subquery)))
+            .or_filter(metadatas::mint_address.eq(any(mint_addresses_subquery)))
```

Comment:
Can some folks audit me on this? I switched 2 subqueries 1 for finding metadata addresses associated to the creators passed in and another for mint_address for owners passed in.

Then use a WHERE address IN (creators_metadata_addressses) OR mint_address IN (mint_addresses)

This query should return metadata that match the attribute filters and either the NFT is created by one of the creators or owned by one of the owners.

Reply:
should work